runner/runner/celery.py

```diff
@@ -11,8 +11,17 @@
 from celery import Celery
 from celery.signals import setup_logging
 
-REDIS_HOST = os.getenv("REDIS_HOST", "localhost")
-REDIS_PORT = os.getenv("REDIS_PORT", "6379")
+broker_workdir = os.getenv("BROKER_WORKDIR", "/tmp")
+
+# Make subdirectory for filesystem broker
+try:
+    if not os.path.exists(os.path.join(broker_workdir, "broker")):
+        os.makedirs(os.path.join(broker_workdir, "broker"))
+except PermissionError:
+    raise PermissionError(
+        f'You do not have permission to create {os.path.join(broker_workdir, "broker")}'
```

Comment:
A note for wherever you move this code to: it probably makes sense to set some other variable to the full path value and then just use that so that you're not calling `os.path.join` over and over again.  Something like:

```python
BROKER_WORKDIR = os.getenv("BROKER_WORKDIR", "/tmp")
BROKER_WORKDIR_DATA = os.path.join(BROKER_WORKDIR, "broker")

#  then later on
os.makdirs(BROKER_WORKDIR_DATA)  # etc, etc
```
It's less about the cost of doing the `os.path.join` and more about have fewer lines to update if we have to change the folder name for some reason.

Also worth pointing out is the all uppercase naming for module level constants.

Reply:
Resolved